NEXT-7952 Merge request body and request parameter

manuelselbach opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?

This is needed to e.g. have the "limit=1" available in the payload for the search of products, etc.

### 2. What does this change do, exactly?

With this change the request parameter and the request body are merged for http methods other than GET.

### 3. Describe each step to reproduce the issue or behaviour.

url: ... "api/v1/search/product?limit=1"
requestbody =
{
"filter" : [

{ "type" : "equals", "field" : "coverId", "value" : null }

]
}

Expected response:

Response of just one procuct:

```
"links" :
{ "first" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=1&page=1", "last" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=1&page=701", "next" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=1&page=2", "self" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=1" }
,
"meta" :
{ "totalCountMode" : 1, "total" : 701 }
,
```

Current response:

Response of 10 products:

```
"links" :
{ "first" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=10&page=1", "last" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=10&page=71", "next" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=10&page=2", "self" : "http://www.ldbs-shop.de.cloud8-vm556.de-nserver.de/api/v1/search/product?limit=1" }
,
"meta" :
{ "totalCountMode" : 1, "total" : 701 }
,
```

### 4. Please link to the relevant issues (if any).

#429 

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopwareBoostDay/platform?pullRequest=435) <br/>All committers have signed the CLA.

manuelselbach committed:
NEXT-7952 - Merge request body and request parameter

With this change the request parameter and the request body are merged
for http methods other than GET. This is needed to e.g. have the
"limit=1" available in the payload for the search of products, etc.

Closes #429

shyim commented:
Hey @manuelselbach ,

We looked again about the original issue and decided that this current behavior is expected. When `POST` HTTP method is used, the limit should be in request body. Sorry about the mistake with the issue. We will try to filter next time better our provided issues.
